src/raycaster-picker.js

```diff
@@ -0,0 +1,29 @@
+import { Raycaster, Vector2 } from 'three';
+
+class RaycasterPicker {
+  constructor(size) {
+    this.raycaster = new Raycaster();
+    this.raycaster.params.Points.threshold = size / 2;
+    this.pointer = new Vector2();
+  }
+
+  pick({ renderer, scene, camera, event }) {
+    const { raycaster, pointer } = this;
+    const { clientX, clientY } = event;
+    const rect = renderer.domElement.getBoundingClientRect();
+
+    pointer.x = ((clientX - rect.left) / (rect.right - rect.left)) * 2 - 1;
+    pointer.y = -((clientY - rect.top) / (rect.bottom - rect.top)) * 2 + 1;
+
+    raycaster.setFromCamera(pointer, camera);
+    const intersects = raycaster.intersectObjects(scene.children, true);
```

Comment:
Random question: is it just the nodes that are `scene.children`, or is it edges too - can edge selection be interfering in the node selection process?